src/realm/bplustree.cpp

```diff
@@ -851,11 +851,12 @@ ref_type BPlusTreeBase::typed_write(ref_type ref, _impl::ArrayWriterBase& out, A
                     // keys (ArrayUnsigned me thinks)
                     Array a(alloc);
                     a.init_from_ref(rot.get_as_ref());
-                    written_node.set_as_ref(j, a.write(out, deep, only_modified, false));
+                    written_node.set_as_ref(j, a.write(out, deep, only_modified, collection_in_mixed));
```

Comment:
I  think we need an explanation specifically for why it is OK to compress here if the flag is set.